core/src/main/java/org/apache/druid/math/expr/Function.java

```diff
@@ -3422,6 +3422,100 @@ ExprEval doApply(ExprEval lhsExpr, ExprEval rhsExpr)
     }
   }
 
+  class ArrayContainsContiguousSubarrayFunction extends ArraysFunction
+  {
+    @Override
+    public String name()
+    {
+      return "array_contains_contiguous_subarray";
+    }
+
+    @Override
+    public boolean hasArrayOutput()
+    {
+      return true;
+    }
+
+    @Nullable
+    @Override
+    public ExpressionType getOutputType(Expr.InputBindingInspector inputTypes, List<Expr> args)
+    {
+      return ExpressionType.LONG;
+    }
+
+    @Override
+    ExprEval doApply(ExprEval lhsExpr, ExprEval rhsExpr)
```

Comment:
The implementation seems to have complexity of O(length1 * length2), which seems suboptimal if we used something like [KMP](https://en.wikipedia.org/wiki/Knuth%E2%80%93Morris%E2%80%93Pratt_algorithm). Maybe something like [Collections#indexOfSublist](https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#indexOfSubList%28java.util.List,%20java.util.List%29) can be better.

Reply:
Hi @LakshSingla, thanks for the comment.
> The implementation seems to have complexity of O(length1 * (length1-length2)), which seems suboptimal if we used something like KMP. Maybe something like Collections#indexOfSublist can be better, but it seems that it might be using the same algorithm underneath. In any case, I think it could be optimized further, and we might not wanna go with the naive implementation if the operands we expect have can have a decent length. String searching

I got your point that you're concerned about the performance of the array function here. Actually, during SQL function translation, it'll try to void necessary array comparing by leveraging a serial of dimension filters which can use Druid's bitmap index to speed up filtering.